specs/packagemanager/PackageManagement.md

```diff
@@ -873,6 +932,17 @@ namespace Microsoft.Windows.Management.Deployment
 
         Boolean IsPackageSetReady(PackageSet packageSet);
 
+        //-------------------------------------------------------------
+        // IsReadyOrNewerAvailable
+
+        // Return true if the package(s) are present and available for use
+
+        PackageReadyOrNewerAvailableStatus IsPackageReadyOrNewerAvailable(String package);
+
+        PackageReadyOrNewerAvailableStatus IsPackageReadyOrNewerAvailableByUri(Windows.Foundation.Uri packageUri);
+
+        PackageReadyOrNewerAvailableStatus IsPackageSetReadyOrNewerAvailable(PackageSet packageSet);
+
```

Comment:
why not just have a IsNewerAvailable* methods we already have IsPackageReady*

Reply:
That would require admin privilege but the ask's for MediumIL processes.

The requirement to only detect newer for registered packages is to prevent an Elevation-of-Privilege. Enumerating packages not registered for the user requires admin privilege, regardless if the user has anything related already registered. PackageManager.FindPackages*() only return packages not registered for the user if called with admin privilege e.g. see 'Remarks' in PackageManager.[FindPackages](https://learn.microsoft.com/uwp/api/windows.management.deployment.packagemanager.findpackagesforuser)() and PackageManager.[FindPackagesForUser](https://learn.microsoft.com/uwp/api/windows.management.deployment.packagemanager.findpackagesforuser)()).

If not for this admin privilege caveat this would be "IsNewerAvailable(criteria)" and non-elevated processes could probe for anything regardless what the user has registered.

Reply:
Another reason to make IsReadyOrNewerAvailable a superset of IsReady rather than add a parallel IsNewerAvailable is performance.

We can't reduce the work done by IsNewerAvailable w/o the EoP issue, but we could return boolean rather than a superset enum. But doing so requires apps also call IsReady to determine what to do e.g.

```
if (!IsReady(x))
    AddPackage(x)
else if IsNewerAvailable(x)
    RegisterPackageFamily(x)
```

NOTE: Ensure*ReadyAsync() does this internally. Dev convenience but doesn't change the perf costs.

We've already been asked to make IsReady faster. As IsReady and IsNewerAvailable implementations heavily overlap, providing the functionality in 1 API provides efficiencies we can't get with separate functions.

Reply:
DISCUSSION: Explanation why IsNewerAvailable can't skip its is-registered logic to compute its answer plus the perf impacts of separate functions vs 1 superset function.